### DISCUSSION TITLE
Container in restart loop for Cyberghost

### DISCUSSION THREAD

**Discussion by oserin** [qdm12/gluetun#516] [Q&A] [answered]
Hi there,

since some days now (I uess about 10) it looks like my container is in a restart loop.
The command that I run to initiate it was:

```sh
docker run -d --name gluetun --cap-add=NET_ADMIN -e VPNSP="Cyberghost" -e REGION="Germany" -e CYBERGHOST_GROUP="Premium UDP Europe" -e USER="myuser" -e PASSWORD="mypassword" -p someport:anotheport -p port3:port3/udp -v /localpath/Gluetun/:/gluetun --restart unless-stopped qmcgaw/private-internet-access
```

I also remember copying client.crt and client.key in my local Gluetun path.

I noticed there was an update since that image therefore I tried now the following but without success 

```sh
docker run -d --name gluetun --cap-add=NET_ADMIN -e VPNSP="Cyberghost" -e REGION="Germany" -e CYBERGHOST_GROUP="Premium UDP Europe" -e OPENVPN_USER="myuser" -e OPENVPN_PASSWORD="mypassword" -p someport:anotheport -p port3:port3/udp -v /localpath/Gluetun/:/gluetun --restart unless-stopped qmcgaw/gluetun
```

How can I debug from there ?

**Comment by qdm12** [comment]
I would need the logs to see why it restarts, thanks !

**Reply by oserin** [reply]
hi qdm12,

and thanks for the support, here is an extract of the last ~2 attempts~ from docker logs gluetun

```
=========================================
================ Gluetun ================
=========================================
==== A mix of OpenVPN, DNS over TLS, ====
======= Shadowsocks and HTTP proxy ======
========= all glued up with Go ==========
=========================================
=========== For tunneling to ============
======== your favorite VPN server =======
=========================================
=== Made with ❤️ by github.com/qdm12 ====
=========================================

Running version latest built on 2021-07-01T18:34:18Z (commit 92212fd)


🔧 Need help? https://github.com/qdm12/gluetun/issues/new
💻 Email? quentin.mcgaw@gmail.com
☕ Slack? Join from the Slack button on Github
💰 Help me? https://github.com/sponsors/qdm12
2021/07/04 20:15:30 INFO Alpine version: 3.14.0
2021/07/04 20:15:30 INFO OpenVPN 2.4 version: 2.4.11
2021/07/04 20:15:30 INFO OpenVPN 2.5 version: 2.5.2
2021/07/04 20:15:30 INFO Unbound version: 1.13.1
2021/07/04 20:15:30 INFO IPtables version: v1.8.7
2021/07/04 20:15:30 INFO Settings summary below:
|--OpenVPN:
   |--Version: 2.5
   |--Verbosity level: 1
   |--Run as root: enabled
   |--Provider:
      |--Cyberghost settings:
         |--Network protocol: udp
         |--Server group: premium udp europe
         |--Regions: germany
         |--Client key is set
         |--Client certificate is set
|--DNS:
   |--Plaintext address: 1.1.1.1
   |--DNS over TLS:
      |--Unbound:
          |--DNS over TLS providers:
              |--Cloudflare
          |--Listening port: 53
          |--Access control:
              |--Allowed:
                  |--0.0.0.0/0
                  |--::/0
          |--Caching: enabled
          |--IPv4 resolution: enabled
          |--IPv6 resolution: disabled
          |--Verbosity level: 1/5
          |--Verbosity details level: 0/4
          |--Validation log level: 0/2
          |--Username: 
      |--Blacklist:
         |--Blocked categories: malicious
         |--Additional IP networks blocked: 13
      |--Update: every 24h0m0s
|--Firewall:
|--System:
   |--Process user ID: 1000
   |--Process group ID: 1000
   |--Timezone: NOT SET ⚠️ - it can cause time related issues
|--HTTP control server:
   |--Listening port: 8000
   |--Logging: enabled
|--Public IP getter:
   |--Fetch period: 12h0m0s
   |--IP file: /tmp/gluetun/ip
|--Github version information: enabled
2021/07/04 20:15:31 INFO storage: merging by most recent 11186 hardcoded servers and 11186 servers read from /gluetun/servers.json
2021/07/04 20:15:31 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/07/04 20:15:31 INFO routing: local ethernet link found: eth0
2021/07/04 20:15:31 INFO routing: local ipnet found: 172.17.0.0/16
2021/07/04 20:15:31 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/07/04 20:15:31 INFO routing: adding route for 0.0.0.0/0
2021/07/04 20:15:31 INFO firewall: firewall disabled, only updating allowed subnets internal list
2021/07/04 20:15:31 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/07/04 20:15:31 INFO openvpn configurator: checking for device /dev/net/tun
2021/07/04 20:15:31 WARN TUN device is not available: open /dev/net/tun: no such file or directory
2021/07/04 20:15:31 INFO openvpn configurator: creating /dev/net/tun
2021/07/04 20:15:31 ERROR open /dev/net/tun: no such device
2021/07/04 20:15:31 INFO Shutdown successful
```

**Reply by qdm12** [reply]
🤔 strange. Try adding the flag `--device=/dev/net/tun` and it *should* work.

What Linux and Docker versions are you running? Find them with

```sh
uname -a
docker version
```

**Reply by oserin** [reply]
Nope i didnt work but I understand this is more a docker / OS issue ?


uname -a : 5.4.102-odroidxu4 #21.02.3 SMP PREEMPT Mon Mar 8 07:55:51 UTC 2021 armv7l armv7l armv7l GNU/Linux

docker version

Client:
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.2-0ubuntu1~20.04.2
 Built:             Tue Mar 30 21:35:24 2021
 OS/Arch:           linux/arm
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.2-0ubuntu1~20.04.2
  Built:            Mon Mar 29 19:10:09 2021
  OS/Arch:          linux/arm
  Experimental:     false
 containerd:
  Version:          1.4.4-0ubuntu1~20.04.2
  GitCommit:        
 runc:
  Version:          1.0.0~rc95-0ubuntu1~20.04.1
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:        



gluetun log

=========================================
================ Gluetun ================
=========================================
==== A mix of OpenVPN, DNS over TLS, ====
======= Shadowsocks and HTTP proxy ======
========= all glued up with Go ==========
=========================================
=========== For tunneling to ============
======== your favorite VPN server =======
=========================================
=== Made with ❤️ by github.com/qdm12 ====
=========================================

Running version latest built on 2021-07-01T18:34:18Z (commit 92212fd)


🔧 Need help? https://github.com/qdm12/gluetun/issues/new
💻 Email? quentin.mcgaw@gmail.com
☕ Slack? Join from the Slack button on Github
💰 Help me? https://github.com/sponsors/qdm12
2021/07/05 06:11:52 INFO Alpine version: 3.14.0
2021/07/05 06:11:52 INFO OpenVPN 2.4 version: 2.4.11
2021/07/05 06:11:52 INFO OpenVPN 2.5 version: 2.5.2
2021/07/05 06:11:52 INFO Unbound version: 1.13.1
2021/07/05 06:11:52 INFO IPtables version: v1.8.7
2021/07/05 06:11:52 INFO Settings summary below:
|--OpenVPN:
   |--Version: 2.5
   |--Verbosity level: 1
   |--Run as root: enabled
   |--Provider:
      |--Cyberghost settings:
         |--Network protocol: udp
         |--Server group: premium udp europe
         |--Regions: germany
         |--Client key is set
         |--Client certificate is set
|--DNS:
   |--Plaintext address: 1.1.1.1
   |--DNS over TLS:
      |--Unbound:
          |--DNS over TLS providers:
              |--Cloudflare
          |--Listening port: 53
          |--Access control:
              |--Allowed:
                  |--0.0.0.0/0
                  |--::/0
          |--Caching: enabled
          |--IPv4 resolution: enabled
          |--IPv6 resolution: disabled
          |--Verbosity level: 1/5
          |--Verbosity details level: 0/4
          |--Validation log level: 0/2
          |--Username: 
      |--Blacklist:
         |--Blocked categories: malicious
         |--Additional IP networks blocked: 13
      |--Update: every 24h0m0s
|--Firewall:
|--System:
   |--Process user ID: 1000
   |--Process group ID: 1000
   |--Timezone: NOT SET ⚠️ - it can cause time related issues
|--HTTP control server:
   |--Listening port: 8000
   |--Logging: enabled
|--Public IP getter:
   |--Fetch period: 12h0m0s
   |--IP file: /tmp/gluetun/ip
|--Github version information: enabled
2021/07/05 06:11:52 INFO storage: merging by most recent 11186 hardcoded servers and 11186 servers read from /gluetun/servers.json
2021/07/05 06:11:52 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/07/05 06:11:52 INFO routing: local ethernet link found: eth0
2021/07/05 06:11:52 INFO routing: local ipnet found: 172.17.0.0/16
2021/07/05 06:11:52 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/07/05 06:11:52 INFO routing: adding route for 0.0.0.0/0
2021/07/05 06:11:52 INFO firewall: firewall disabled, only updating allowed subnets internal list
2021/07/05 06:11:52 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/07/05 06:11:52 INFO openvpn configurator: checking for device /dev/net/tun
2021/07/05 06:11:52 WARN TUN device is not available: open /dev/net/tun: no such device
2021/07/05 06:11:52 INFO openvpn configurator: creating /dev/net/tun
2021/07/05 06:11:52 ERROR file exists
2021/07/05 06:11:52 INFO Shutdown successful

**Reply by oserin** [reply]
Ok a reboot fixed my issue :o , it seems an update of the firmware caused it. Thx for the help

**Comment by oserin** [comment, marked as answer]
Reboot in case of kernel/firmware update did the trick
